service/simpledb/unmarshall_error.go

```diff
@@ -0,0 +1,51 @@
+package simpledb
+
+import (
+	"encoding/xml"
+	"io"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/request"
+)
+
+type xmlErrorDetail struct {
+	Code    string `xml:"Code"`
+	Message string `xml:"Message"`
+}
+
+type xmlErrorResponse struct {
+	XMLName   xml.Name         `xml:"Response"`
+	Errors    []xmlErrorDetail `xml:"Errors>Error"`
+	RequestID string           `xml:"RequestID"`
+}
+
+func unmarshalError(r *request.Request) {
```

Comment:
You are spot on, I did base this code on the S3 code, with the main difference being that the SimpleDB error responses is different from the S3 error responses. (Note that the SDB code has `xmlErrorDetail` and `xmlErrorResponse` types, while the S3 code only has the `xmlErrorResponse` type).

There are also some other differences:

* the S3 code has some S3-specific checks for incorrect region
* the SDB error response can contain multiple error codes
* there are extra serialization error conditions when checking the SDB error response

It just seemed a bit too tricky to make the same bit of code work for both S3 and SDB, and then there was the issue of what package to put the common code in.

So  I have not made any changes to this code at this time, but am happy to in future if you want.